docs/TimerMocks.md

```diff
@@ -3,176 +3,174 @@ id: timer-mocks
+  jest.useFakeTimers();
+  // ...
+});
 
-    // And it should have created a new timer to start the game over in
-    // 10 seconds
-    expect(setTimeout).toHaveBeenCalledTimes(2);
-    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 10000);
-  });
+test('do something with real timers', () => {
+  // ...
 });
 ```
 
-## Advance Timers by Time
+Here we use an `afterEach` to call `jest.useRealTimers()` after every single
+test.
 
-Another possibility is use `jest.advanceTimersByTime(msToRun)`. When this API is called, all timers are advanced by `msToRun` milliseconds. All pending "macro-tasks" that have been queued via setTimeout() or setInterval(), and would be executed during this time frame, will be executed. Additionally, if those macro-tasks schedule new macro-tasks that would be executed within the same time frame, those will be executed until there are no more macro-tasks remaining in the queue that should be run within msToRun milliseconds.
+This pattern can help keep behavior predictable as your test file
+grows.
 
-```javascript
-// timerGame.js
-'use strict';
+- It will establish a clear baseline ->
+"timers are real unless otherwise is set".
+- The state of the fake timers is always reset between tests, since all tests
+wanting to use them have to call `jest.useFakeTimers()` first.
 
-function timerGame(callback) {
-  console.log('Ready....go!');
-  setTimeout(() => {
-    console.log("Time's up -- stop!");
-    callback && callback();
-  }, 1000);
-}
+## Handling recursive timers
 
-module.exports = timerGame;
-```
+Scenarios exist where you might have a recursive timer -- that is a timer
+setting a new timer with itself as the callback.
 
-```javascript
-it('calls the callback after 1 second via advanceTimersByTime', () => {
-  const timerGame = require('../timerGame');
-  const callback = jest.fn();
+If you use `jest.runAllTimers()` to advance time, it will execute all pending
+tasks. If those tasks themselves schedule new tasks, those will be continually
+exhausted until there are no more tasks remaining in the queue. So with a
+recursive timer, you would end up with an endless loop.
 
-  timerGame(callback);
+To solve this, Jest ships with an alternative called
+`jest.runOnlyPendingTimers()`. This will run only the tasks queued by
+`setTimeout()` or `setInterval()` up until that point.
 
-  // At this point in time, the callback should not have been called yet
-  expect(callback).not.toBeCalled();
+TODO: Recursive example here
 
-  // Fast-forward until all timers have been executed
-  jest.advanceTimersByTime(1000);
+## More code examples
 
-  // Now our callback should have been called!
-  expect(callback).toBeCalled();
-  expect(callback).toHaveBeenCalledTimes(1);
-});
-```
+TODO: Info about / link to more code examples here.
+
+## Frequently Asked Questions
+
+### Can I spy on `setTimeout()` while using fake timers?
+
+The spesific timer function used is an implementation detail, and it would
```

Comment:
Thanks, updated the spelling. 👍 

As for answering the question directly; yes, you are right, I agree. I will try to figure out a way of wording it. 

It is not a feature in the new timer implementation to support it, so I'm inclined to say "no" even if there were to exist 'hacky' ways of doing it (I haven't tested this exhaustively yet). As either way it seems like bad practise to me.

Reply:
I think re-wording the question itself is most likely the way to go in order to be able to provide a direct and reasonably uncomplicated answer.

What do you think about this instead:

----

### Jest says `setTimeout()` is no longer a mock or spy function, how can I assert against it?

In the old ("legacy") fake timers, the native timer functions were replaced by Jest mock functions. After moving to the new ("modern") fake timers based on an [external library](https://github.com/sinonjs/fake-timers) in Jest 27, this is no longer a feature.

The specific timer function used (e.g. `setTimeout()`) should be considered an implementation detail, and it would generally be more robust to test against what you expect to happen when a task runs.

Looking back at the first code example on this page, you will see that we assert if the mocked callback function is called with the parameter we expect. This decouples us from the details of what happens inside `timerTest()`. At some point in the future, we can safely replace `setTimeout()` with another timer implementation (for example `setImmediate()`) without invalidating the test.

If you need to verify that callbacks are scheduled with a particular time or interval, consider using `jest.advanceTimersByTime()` and make assertions based on what you expect at different points in time.

----

An added bonus is that the phrase `a mock or spy function` is part of the error message users would get if they attempt to assert against one of the native timer functions with modern fake timers. So this could also have SEO benefits.

Reply:
Perhaps... "(...) this is no longer a feature. So the short answer is that you can't... at least not directly."

Reply:
FYI: Preview is live with an [updated question and answer](https://deploy-preview-11731--jestjs.netlify.app/docs/timer-mocks#timer-functions-are-no-longer-a-mock-or-spy-function-how-can-i-assert-against-them). In order to give a more direct answer, I found it necessary to rephrase the question itself. As an added bonus, the phrase `a mock or spy function` is part of the error message users would see when trying to assert against native timer functions with the modern fake timers. It's possible that this could have search-engine benefits for users googling the error.